packages/plugin-react-navigation/react-navigation.js

```diff
@@ -0,0 +1,57 @@
+const React = require('react')
+
+class Plugin {
+  constructor () {
+    this.name = 'reactNavigation'
+  }
+
+  load (client) {
+    const createNavigationContainer = (NavigationContainer) => React.forwardRef((props, ref) => {
+      const { onReady, onStateChange, ...forwardProps } = props
+      const navigationRef = ref || React.useRef(null)
+      const routeNameRef = React.useRef(undefined)
+
+      const wrappedOnReady = (...args) => {
+        const currentRoute = navigationRef.current ? navigationRef.current.getCurrentRoute() : null
+        if (currentRoute) {
+          routeNameRef.current = currentRoute.name
+          client.setContext(currentRoute.name)
```

Comment:
Should this leave a bradcrumb too? I'm not sure what the lifecycle is like so it might not be necessary if there's an `onStateChange` soon after, but it might be useful to see the initial route in case that crashes before an `onStateChange` — e.g. if a user isn't logged in they'd see the login screen, but otherwise would have a different view and it might be useful for that to be reflected in the breadcrumbs

React Native Navigation calls `componentDidAppear` for the initial route, so you get a breadcrumb for the first rendered view with that plugin